ddl/ddl_api.go

```diff
@@ -2203,7 +2204,11 @@ func (d *ddl) AddTablePartitions(ctx sessionctx.Context, ident ast.Ident, spec *
 	// old partitions to check all partitions is strictly increasing.
 	tmp := *partInfo
 	tmp.Definitions = append(pi.Definitions, tmp.Definitions...)
-	err = checkCreatePartitionValue(ctx, meta, &tmp, t.Cols())
+	if len(tmp.Columns) == 0 {
+		err = checkCreatePartitionValue(ctx, meta, &tmp, t.Cols())
+	} else {
+		err = checkRangeColumnsPartitionValue(ctx, meta, &tmp)
+	}
```

Comment:
Do we need to consider `checkPartitionByRange` and `checkPartitionByRangeColumn`?

Reply:
Great, already refactor the code.